### PR TITLE
Adds RootVolumeSize Parameter

### PR DESCRIPTION
**Pull request by userhas404d**
Adds parameter to manually set the size of the root partition.

Caveat: Default EC2 instance root volume size is determined by the size of the snapshot in use by the AMI. Default values assume only SPEL and Amazon provided Windows AMIs are being used with these templates.

### CODE REVIEW

**Comment by userhas404d** [comment]
go codebuild go

**Comment by userhas404d** [comment]
go codebuild go

**Comment by userhas404d** [comment]
go codebuild go

File: modules/lx-autoscale/watchmaker-lx-autoscale.template.cfn.yaml [hunk 1]
```diff
@@ -881,6 +894,16 @@ Resources:
 
               #!/bin/bash -xe
 
+              # Extend any available LVM PVs
+              if [[ -x $( which pvs ) ]]
+              then
+                LVMPVS=($(pvs --noheadings -o pv_name))
```
Comment (lorengordon) [inline modules/lx-autoscale/watchmaker-lx-autoscale.template.cfn.yaml]: Mind providing an example of this output as a pr comment, so can get a better idea of what this is doing?

File: modules/lx-autoscale/watchmaker-lx-autoscale.template.cfn.yaml [hunk 2]
```diff
@@ -949,7 +973,14 @@ Resources:
                         - [${local_SupportsNvme}, ${AppVolumeMountPath}]
                   - local_SupportsNvme: !If [SupportsNvme, /dev/nvme1n1, /dev/xvdf]
                 - ''
-              local_AssignInstanceRole: !If [AssignInstanceRole, !Sub '--role ${InstanceRole}', '']
+              local_ExpandRootVolume: !If 
+                - ExpandRootVolume
+                - |+
+                  growpart:
+                    mode: auto
+                    devices: [ '/dev/xvda', '/dev/xvda2', '/dev/nvme0n1', '/dev/nvme0n1p2' ]
```
Comment (lorengordon) [inline modules/lx-autoscale/watchmaker-lx-autoscale.template.cfn.yaml]: What do the partition's end up looking like, in the case of a successful root volume expansion?

File: modules/lx-autoscale/watchmaker-lx-autoscale.template.cfn.yaml (hunk 1 above)
Comment (userhas404d) [reply modules/lx-autoscale/watchmaker-lx-autoscale.template.cfn.yaml]: [Updated](https://github.com/plus3it/terraform-aws-watchmaker/pull/92/commits/4b7d3d11da4a67b15d8ed754fb78f3d5fd1c98ac)

File: modules/lx-autoscale/watchmaker-lx-autoscale.template.cfn.yaml (hunk 2 above)
Comment (userhas404d) [reply modules/lx-autoscale/watchmaker-lx-autoscale.template.cfn.yaml]: On the linux side it's still up to the user to expand the partitions but here's a breakdown on what LVM and the partitions look like: 

```
df -h 
Filesystem                       Size  Used Avail Use% Mounted on
/dev/mapper/VolGroup00-rootVol   3.9G  2.1G  1.6G  58% /
devtmpfs                         471M     0  471M   0% /dev
tmpfs                            495M     0  495M   0% /dev/shm
tmpfs                            495M  512K  495M   1% /run
tmpfs                            495M     0  495M   0% /sys/fs/cgroup
tmpfs                            495M     0  495M   0% /tmp
/dev/mapper/VolGroup00-homeVol   976M  2.6M  907M   1% /home
/dev/mapper/VolGroup00-varVol    2.0G  198M  1.6G  11% /var
/dev/xvda1                       453M  118M  308M  28% /boot
/dev/mapper/VolGroup00-logVol    2.0G   16M  1.8G   1% /var/log
/dev/mapper/VolGroup00-auditVol  8.3G   37M  7.8G   1% /var/log/audit
tmpfs                             99M     0   99M   0% /run/user/1000

sudo lvs
LV       VG         Attr       LSize Pool Origin Data%  Meta%  Move Log Cpy%Sync Convert
auditVol VolGroup00 -wi-ao---- 8.53g
homeVol  VolGroup00 -wi-ao---- 1.00g
logVol   VolGroup00 -wi-ao---- 2.00g
rootVol  VolGroup00 -wi-ao---- 4.00g
swapVol  VolGroup00 -wi-ao---- 2.00g
varVol   VolGroup00 -wi-ao---- 2.00g

sudo vgs
VG         #PV #LV #SN Attr   VSize  VFree
VolGroup00   1   6   0 wz--n- 39.53g 20.00g

sudo pvs
PV         VG         Fmt  Attr PSize  PFree
/dev/xvda2 VolGroup00 lvm2 a--  39.53g 20.00g


[maintuser@ip-172-31-60-104 ~]$ sudo lvextend -L+5G /dev/VolGroup00/rootVol
    Size of logical volume VolGroup00/rootVol changed from 4.00 GiB (1024 extents) to 9.00 GiB (2304 extents).
    Logical volume VolGroup00/rootVol successfully resized.
[maintuser@ip-172-31-60-104 ~]$ sudo resize2fs /dev/VolGroup00/rootVol
    resize2fs 1.42.9 (28-Dec-2013)
    Filesystem at /dev/VolGroup00/rootVol is mounted on /; on-line resizing required
    old_desc_blocks = 1, new_desc_blocks = 2
    The filesystem on /dev/VolGroup00/rootVol is now 2359296 blocks long.


df -h
Filesystem                       Size  Used Avail Use% Mounted on
/dev/mapper/VolGroup00-rootVol   8.8G  2.1G  6.3G  25% /
devtmpfs                         471M     0  471M   0% /dev
tmpfs                            495M     0  495M   0% /dev/shm
tmpfs                            495M  568K  495M   1% /run
tmpfs                            495M     0  495M   0% /sys/fs/cgroup
tmpfs                            495M     0  495M   0% /tmp
/dev/mapper/VolGroup00-varVol    2.0G  198M  1.6G  11% /var
/dev/mapper/VolGroup00-homeVol   976M  2.6M  907M   1% /home
/dev/xvda1                       453M  118M  308M  28% /boot
/dev/mapper/VolGroup00-logVol    2.0G   16M  1.8G   1% /var/log
/dev/mapper/VolGroup00-auditVol  8.3G   37M  7.8G   1% /var/log/audit
tmpfs                             99M     0   99M   0% /run/user/1000
tmpfs                             99M     0   99M   0% /run/user/0
```

Windows is completely seamless and `C` will auto-expand to take advantage of all available space.

**Comment by lorengordon** [comment]
Gotcha. This looks good, and will work for the "launch" use case, but it will not expand the partition for the "update" use case. I think we should move the logic into the cfn-init config-sets, so it supports both use cases.

**Comment by userhas404d** [comment]
go codebuild go

**Comment by lorengordon** [comment]
Needs a rebase please...

**Comment by userhas404d** [comment]
go codebuild go

**Comment by lorengordon** [comment]
go codebuild go

**Pull request merged**